docs/src/model_setup/forcing_functions.md

```diff
@@ -277,3 +280,88 @@ ContinuousForcing{Nothing} at (Center, Center, Center)
 └── field dependencies: (:T,)
 ```
 
+## `AdvectiveForcing`
+
+`AdvectiveForcing` defines a forcing function that represents advection by
+a separate or "slip" velocity relative to the prognostic model velocity field.
+`AdvectiveForcing` is implemented with native Oceananigans advection operators,
+which means that tracers advected by the "flux form" advection term
+``∇ ⋅ u⃗ˢˡⁱᵖ c``. Caution is advised when ``u⃗ˢˡⁱᵖ`` is not divergence free.
+
+As an example, consider a model for sediment settling at a constant rate:
+
+```jldoctest
+using Oceananigans
+
+r_sediment = 1e-4 # "Fine sand"
+ρ_sediment = 1200 # kg m⁻³
+ρ_ocean = 1026 # kg m⁻³
+Δb = 9.81 * (ρ_ocean - ρ_sediment) / ρ_ocean
+ν_molecular = 1.05e-6
+w_sediment = 2/9 * Δb / ν_molecular * r_sediment^2
+
+sinking = AdvectiveForcing(UpwindBiasedFifthOrder(), w=w_sediment)
+
+# output
+AdvectiveForcing with the UpwindBiasedFifthOrder scheme:
+├── u: ZeroField{Int64}
+├── v: ZeroField{Int64}
+└── w: ConstantField(-0.00352102)
+```
+
+The first argument to `AdvectiveForcing` is the advection scheme (here `WENO5()`).
+The three keyword arguments specify the `u`, `v`, and `w` components of the separate
+slip velocity field. The default for each `u, v, w` is `ZeroField`.
+
+Next we consider a dynamically-evolving slip velocity. For this we use `ZFaceField`
+with appropriate boundary conditions as our slip velocity:
+
+```jldoctest sinking
+using Oceananigans
+using Oceananigans.BoundaryConditions: ImpenetrableBoundaryCondition
+
+grid = RectilinearGrid(size=(32, 32, 32), x=(-10, 10), y=(-10, 10), z=(-4, 4),
+                       topology=(Periodic, Periodic, Bounded))
+
+no_penetration = ImpenetrableBoundaryCondition()
+slip_bcs = FieldBoundaryConditions(grid, (Center, Center, Face),
+                                   top=no_penetration, bottom=no_penetration)
+
+w_slip = ZFaceField(grid, boundary_conditions=slip_bcs)
+sinking = AdvectiveForcing(WENO5(; grid), w=w_slip)
+
+# output
+AdvectiveForcing with the WENO5 scheme:
+├── u: ZeroField{Int64}
+├── v: ZeroField{Int64}
+└── w: 32×32×33 Field{Center, Center, Face} on RectilinearGrid on CPU
+```
+
+To compute the slip velocity, we must add a `Callback`to `simulations.callback` that
+computes `w_slip` ever iteration:
+
+```jldoctest sinking
+using Oceananigans.BoundaryConditions: fill_halo_regions!
+
+model = NonhydrostaticModel(; grid, tracers=(:b, :P), forcing=(; P=sinking))
+simulation = Simulation(model; Δt=1, stop_iteration=100)
+
+# Build abstract operation for slip velocity
+b_particle = - 1e-4 # relative buoyancy depends on reference density and initial buoyancy condition
+b = model.tracers.b
```

Comment:
Is that any difference here if we define `b = model.tracers.b` or `b = simulation.model.tracers.b`?

Reply:
If `simulation` was built with `model`, then they are the same.